app/src/main/java/com/github/geohunt/app/model/database/firebase/FirebaseLoggedUserContext.kt

```diff
@@ -0,0 +1,254 @@
+
+        val challengeEntry = ChallengeEntry(
+            authorId = loggedUserRef.id,
+            publishedDate = DateUtils.utcIso8601Now(),
+            expirationDate = DateUtils.utcIso8601FromLocalNullable(expirationDate),
+            claims = mapOf(),
+            location = location
+        )
+
+        // Get the reference to the thumbnail Bitmap and set the value
+        val thumbnailBitmap = database.getChallengeThumbnailById(challengeId)
+        thumbnailBitmap.value = thumbnail
+
+        // Create both jobs (update database, update storage)
+        val submitToDatabaseTask = dbChallengeRef.setValue(challengeEntry)
+        val submitToStorageTask = thumbnailBitmap.saveToLocalStorageThenSubmit()
+        val pushChallengeToUserTask = database.dbUserRef.child(loggedUserRef.id)
+            .child("challenges")
+            .child(challengeId).setValue(true)
+
+        // Finally make the completable task that succeed if both task succeeded
+        return Tasks.whenAll(submitToDatabaseTask, submitToStorageTask, pushChallengeToUserTask).thenMap {
+            FirebaseChallenge(
+                cid = challengeId,
+                author = loggedUserRef,
+                thumbnail = thumbnailBitmap,
+                publishedDate = DateUtils.localFromUtcIso8601(challengeEntry.publishedDate!!),
+                expirationDate = expirationDate,
+                correctLocation = location,
+                claims = listOf()
+            )
+        }
+    }
+
+    override fun Challenge.submitClaim(thumbnail: Bitmap, location: Location): Task<Claim> {
+        require(thumbnail.width * thumbnail.height < R.integer.maximum_number_of_pixel_per_photo)
+
+        // State variable
+        val dbChallengeRef = database.dbChallengeRef.getChallengeRefFromId(cid)
+        val dbClaimRef = database.dbClaimRef.child(cid).push()
+        val claimId = cid + "!!" + dbClaimRef.key!!
+
+        val claimEntry = ClaimEntry(
+            user = loggedUserRef.id,
+            cid = cid,
+            time = DateUtils.utcIso8601Now(),
+            location = location
+        )
+
+        // Get the reference to the thumbnail Bitmap and set the value
+        val thumbnailBitmap = database.getClaimThumbnailById(claimId)
+        thumbnailBitmap.value = thumbnail
+
+        // Create both jobs (update database, update storage, update claim list)
+        val submitToDatabaseTask = dbClaimRef.setValue(claimEntry)
+        val submitToStorageTask = thumbnailBitmap.saveToLocalStorageThenSubmit()
+        val pushChallengeListTask = dbChallengeRef.child("claims").child(claimId).setValue(true)
+
+        // Finally make the completable task that succeed if both task succeeded
+        return Tasks.whenAll(submitToDatabaseTask, submitToStorageTask, pushChallengeListTask).thenMap {
+            FirebaseClaim(
+                id = claimId,
+                user = loggedUserRef,
+                time = DateUtils.localFromUtcIso8601(claimEntry.time!!),

```

Comment:
The non-null assertion (!!) on a non-null receiver of type String is unnecessary. In other words, remove the `!!`.